### ISSUE TITLE
Add "Zawinski's Law"

### ISSUE THREAD

**Issue by dwmkerr** [dwmkerr/hacker-laws#309]
> Every program attempts to expand until it can read mail. Those programs which cannot so expand are replaced by ones which can.

https://en.wikipedia.org/wiki/Jamie_Zawinski

**Comment by troyane** [comment]
At first look it sounds to me as a joke. Then, checked Wiki. Found that Zawinski himself called it the "Law of Software Envelopment". Eric S. Raymond commented that while this law goes against the minimalist philosophy of Unix (a set of "small, sharp tools"), it actually addresses the real need of end users to keep together tools for interrelated tasks, even though for a coder implementation of these tools clearly consists of independent jobs.

Zawinski law could be treated as "Feature creep" (https://en.wikipedia.org/wiki/Feature_creep).

@dwmkerr This law should definitely go well explained.

**Comment by harvey-green** [comment]
I think this law could be brought up to date a bit by changing it to "expand until it uses a Machine Learning API" or "expand until it can post to your Facebook timeline".

**Comment by dwmkerr** [comment]
@troyane agreed it would need explanation, but I think it could be a good one, particularly with feature creep as a reference!
